include/highfive/bits/H5PropertyList_misc.hpp

```diff
@@ -342,10 +353,69 @@ inline const std::vector<hsize_t>& Chunking::getDimensions() const noexcept {
     return _dims;
 }
 
+inline void Chunking::apply(const hid_t hid) const {
+    if (H5Pset_chunk(hid, static_cast<int>(_dims.size()), _dims.data()) < 0) {
+        HDF5ErrMapper::ToException<PropertyException>("Error setting chunk property");
+    }
+}
+
 template <typename... Args>
 inline Chunking::Chunking(hsize_t item, Args... args)
     : Chunking(std::vector<hsize_t>{item, static_cast<hsize_t>(args)...}) {}
 
+inline std::vector<std::size_t> Chunking::guessChunkingSize(
+    const std::vector<std::size_t>& dims,
+    const std::vector<std::size_t>& max_dims,
+    std::size_t typesize) {
+    const std::size_t CHUNK_BASE = 16 * 1024;   // Multiplier by which chunks are adjusted
+    const std::size_t CHUNK_MIN = 8 * 1024;     // Soft lower limit (8k)
+    const std::size_t CHUNK_MAX = 1024 * 1024;  // Hard upper limit (1M)
+
+    std::vector<std::size_t> chunkingDims = dims;
+    // If the dimension is unlimited, set chunksize to 1024 along that
+    for (std::size_t i = 0; i < dims.size(); i++) {
+        if (max_dims[i] == SIZE_MAX) {
+            chunkingDims[i] = 1024;
+        }
+    }
+
+    std::size_t dset_size = details::compute_total_size(chunkingDims) * typesize;
+    double target_size = CHUNK_BASE *
+                         std::exp2(std::log10(static_cast<double>(dset_size) / (1024. * 1024.)));
+
+    if (target_size > CHUNK_MAX) {
+        target_size = CHUNK_MAX;
+    } else if (target_size < CHUNK_MIN) {
+        target_size = CHUNK_MIN;
+    }
+
+    std::size_t idx = 0;
+    while (1) {
+        // Repeatedly loop over the axes, dividing them by 2.  Stop when:
+        // 1a. We're smaller than the target chunk size, OR
+        // 1b. We're within 50% of the target chunk size, AND
+        //  2. The chunk is smaller than the maximum chunk size
+
+        std::size_t chunk_size = details::compute_total_size(chunkingDims) * typesize;
+
+        if ((static_cast<double>(chunk_size) < target_size ||
+             std::abs(static_cast<double>(chunk_size) - target_size) / target_size < 0.5) &&
+            chunk_size < CHUNK_MAX) {
+            break;
+        }
+
+        if (details::compute_total_size(chunkingDims) == 1) {
+            break;  // Element size larger than CHUNK_MAX
+        }
+
+        chunkingDims[idx % chunkingDims.size()] = static_cast<std::size_t>(
+            std::ceil(static_cast<double>(chunkingDims[idx % chunkingDims.size()]) / 2.));
```

Comment:
The converse of wide table would be very narrow tables, e.g. as they occur in FVM simulations. In computation fluid dynamics (depending on the equation), but commonly there's `2`, ..., `8` conserved variables. Hence, one could easily have a shape `(nt, nx, ny, nz, 5)`, with moderately large `nx`, `ny`, `nz`; and unlimited `nt`. We might end up dividing the `5` by `2`, resulting in a chunk shape of `(ct, cx, cy, cz, 3)`. To me this is reason to be skeptical because: a) the chunk size is not a power of 2; and b) we'll need to chunks to fetch the conserved variables.

The power of two issue I feel is awkward. Note, that it occurs quite generally. Whenever one of `dims[i]` isn't a power of two, the chunk size will likely not be either.

Data is read efficiently from disk and RAM in contiguous chunks of a power of 2 size. In page allocated files, the chunks won't fill the page, leaving a gap at the end. This gap can either be filled by small (chunks of) datasets, causing fragmentation of the file; or left empty. These chunks might not fit very nicely into caches.

The size inflation issue is maybe less of a concern. For `(n, 3, 3)` is 1.7x larger; for `(n, 3, 3, 3, 3)` is 3.1x (maybe that's too special of a case). In the context of compression, it seems unfortunate to start off with a 1.7x larger dataset. Yes, there's a good chance that all the extra space will be compressed very efficiently.

I'm not fully sure how max-dims works; and I need to check this. The concern is that someone would pad their dims to get powers of two, e.g. `dims = (n, 3)` with `max_dims=(n, 4)`. This algorithm would often create chunks of shape `(n, 3)`. Which has the same two issues, but this time it feels like the user specifically did something to get powers of two; but the choice of chunk undoes that work. Furthermore, the inflation aspect is more pronounced this time, it can be reasonably close to `2**d` where `d` is the number of padded dimensions. Fortunately, I think the increase in size would only happen if one of the "padded" dimensions is extended.